File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0006-join/type/TEmployeeTable.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinitionInterface.ftl", "tEmployeeTable"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public interface TEmployeeTable extends com.gs.dmn.runtime.DMNType {
    static TEmployeeTable toTEmployeeTable(Object other) {
        if (other == null) {
            return null;
        } else if (TEmployeeTable.class.isAssignableFrom(other.getClass())) {
            return (TEmployeeTable)other;
        } else if (other instanceof com.gs.dmn.runtime.Context) {
            TEmployeeTableImpl result_ = new TEmployeeTableImpl();
            result_.setId((String)((com.gs.dmn.runtime.Context)other).get("id"));
            result_.setName((String)((com.gs.dmn.runtime.Context)other).get("name"));
            result_.setDeptNum((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("deptNum"));
            return result_;
        } else if (other instanceof com.gs.dmn.runtime.DMNType) {
            return toTEmployeeTable(((com.gs.dmn.runtime.DMNType)other).toContext());
        } else {
            throw new com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.getClass().getSimpleName(), TEmployeeTable.class.getSimpleName()));
        }
    }

    @com.fasterxml.jackson.annotation.JsonGetter("id")
    String getId();

    @com.fasterxml.jackson.annotation.JsonGetter("name")
    String getName();

    @com.fasterxml.jackson.annotation.JsonGetter("deptNum")
    java.math.BigDecimal getDeptNum();

    default com.gs.dmn.runtime.Context toContext() {
        com.gs.dmn.runtime.Context context = new com.gs.dmn.runtime.Context();
        context.put("id", getId());
        context.put("name", getName());
        context.put("deptNum", getDeptNum());
        return context;
    }

    default boolean equalTo(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TEmployeeTable other = (TEmployeeTable) o;
        if (this.getDeptNum() != null ? !this.getDeptNum().equals(other.getDeptNum()) : other.getDeptNum() != null) return false;
        if (this.getId() != null ? !this.getId().equals(other.getId()) : other.getId() != null) return false;
        if (this.getName() != null ? !this.getName().equals(other.getName()) : other.getName() != null) return false;

        return true;
    }

    default int hash() {
        int result = 0;
        result = 31 * result + (this.getDeptNum() != null ? this.getDeptNum().hashCode() : 0);
        result = 31 * result + (this.getId() != null ? this.getId().hashCode() : 0);
        result = 31 * result + (this.getName() != null ? this.getName().hashCode() : 0);
        return result;
    }

    default String asString() {
        StringBuilder result_ = new StringBuilder("{");
        result_.append("deptNum=" + getDeptNum());
        result_.append(", id=" + getId());
        result_.append(", name=" + getName());
        result_.append("}");
        return result_.toString();
    }
}

File: dmn-test-cases/tck/1.1/cl3/0006-join/translator/expected/kotlin/dmn/type/TEmployeeTable.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinitionInterface.ftl", "tEmployeeTable"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(`as` = type.TEmployeeTableImpl::class)
interface TEmployeeTable : com.gs.dmn.runtime.DMNType {
    @get:com.fasterxml.jackson.annotation.JsonGetter("id")
    val id: String?

    @get:com.fasterxml.jackson.annotation.JsonGetter("name")
    val name: String?

    @get:com.fasterxml.jackson.annotation.JsonGetter("deptNum")
    val deptNum: java.math.BigDecimal?

    override fun toContext(): com.gs.dmn.runtime.Context {
        val context = com.gs.dmn.runtime.Context()
        context.put("id", this.id)
        context.put("name", this.name)
        context.put("deptNum", this.deptNum)
        return context
    }

    fun equalTo(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as TEmployeeTable
        if (if (this.deptNum != null) this.deptNum != other.deptNum else other.deptNum != null) return false
        if (if (this.id != null) this.id != other.id else other.id != null) return false
        if (if (this.name != null) this.name != other.name else other.name != null) return false

        return true
    }

    fun hash(): Int {
        var result = 0
        result = 31 * result + (if (this.deptNum != null) this.deptNum.hashCode() else 0)
        result = 31 * result + (if (this.id != null) this.id.hashCode() else 0)
        result = 31 * result + (if (this.name != null) this.name.hashCode() else 0)
        return result
    }

    fun asString(): String {
        val result_ = StringBuilder("{")
        result_.append("deptNum=" + deptNum)
        result_.append(", id=" + id)
        result_.append(", name=" + name)
        result_.append("}")
        return result_.toString()
    }

    companion object {
        @JvmStatic
        fun toTEmployeeTable(other: Any?): TEmployeeTable? {
            if (other == null) {
                return null
            } else if (other is TEmployeeTable?) {
                return other
            } else if (other is com.gs.dmn.runtime.Context) {
                var result_ = TEmployeeTableImpl()
                result_.id = other.get("id") as String?
                result_.name = other.get("name") as String?
                result_.deptNum = other.get("deptNum") as java.math.BigDecimal?
                return result_
            } else if (other is com.gs.dmn.runtime.DMNType) {
                return toTEmployeeTable(other.toContext())
            } else {
                throw com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.javaClass.getSimpleName(), TEmployeeTable::class.java.getSimpleName()))
            }
        }
    }
}
